java/src/main/native/src/TableJni.cpp

```diff
@@ -2635,7 +2637,13 @@ JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_Table_convertCudfToArrowTable(JNIEnv
     // The pointer to the shared_ptr<> is returned as a jlong.
     using result_t = std::shared_ptr<arrow::Table>;
 
-    auto result = cudf::to_arrow(*tview, state->get_column_metadata(*tview));
+    auto got_arrow_schema   = cudf::to_arrow_schema(*tview, state->get_column_metadata(*tview));
+    got_arrow_schema->flags = ARROW_FLAG_NULLABLE;
```

Comment:
I think the issue is that you are only setting the top level schema to be nullable.  All child columns must also be nullable as they too can change, and their children, etc....

Reply:
Ah of course, apologies for the oversight. I hadn't constructed a test for nested types. Let me try that now.

Reply:
Pushed a version that sets this flag recursively.